scripts/train_alternating.py

```
@@ -0,0 +1,281 @@
+    checkpoint = {
+        'model_state_dict': model.state_dict(),
+        'optimizer_state_dict': optimizer.state_dict(),
+        'step': step,
+        'loss': loss,
+    }
+    checkpoint_path = os.path.join(checkpoint_dir, f'checkpoint_step_{step}.pt')
+    torch.save(checkpoint, checkpoint_path)
+    return checkpoint_path
+
+
+def main():
```

Pylint (3.3.7)

[refactor] Too many local variables (35/15)

(R0914)